onmt/inputters/inputter.py

```diff
@@ -288,6 +287,17 @@ def _build_field_vocab(field, counter, size_multiple=1, **kwargs):
         _pad_vocab_to_multiple(field.vocab, size_multiple)
 
 
+def _update_field_vocab(field, old_field):
+    new_vocabulary = []
+    for w in field.vocab.itos:
+        if w not in old_field.vocab.stoi:
+            new_vocabulary.append(w)
+    logger.info('New vocabulary has %d new tokens.' % (len(new_vocabulary)))
+    print(new_vocabulary)
+    old_field.vocab.extend(field.vocab)
+    field.vocab = old_field.vocab
```

Comment:
What happens here in terms of `freqs` for the new tokens? E.g. if we would like to add a few known tokens to the vocab, without necessarily calling `build_vocab` again, how would that be handled? Should we always have some 'licit' frequency in the new vocab?

Reply:
The main use case I was thinking is finetuning a previously trained model with a new corpus which potentially has new tokens in vocabulary (E.g. new protected tokens for domain tagging). In such cases we need to add these tokens to the vocabulary accepted by the model and thus build vocabulary would be run for the new corpus. 
I don't know exactly what you mean by the `freqs` for the new tokens. As far as I know, token frequencies are just to remove those which are below a given frequency (E.g. `src_vocab_threshold`). Which would be a use case in which it wouldn't be necessary to call build_vocab or have a 'licit' frequency?